### PR TITLE
Charlesmchen/swift4

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: Signal.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -4431,7 +4439,8 @@
 				PROVISIONING_PROFILE = "";
 				SWIFT_OBJC_BRIDGING_HEADER = "Signal/test/SignalTests-Bridging-Header.h";
 				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
-				SWIFT_VERSION = 3.0;
+				SWIFT_SWIFT3_OBJC_INFERENCE = On;
```
Comment (charlesmchen-signal) [inline Signal.xcodeproj/project.pbxproj]: > SWIFT_SWIFT3_OBJC_INFERENCE 

We might want to turn this off? 

> SWIFT_VERSION

Not sure if we should use 4.0 or 4.1.

File: Signal/src/call/CallService.swift [hunk 2]
```diff
@@ -154,17 +154,23 @@ private class SignalCallData: NSObject {
 
         let (callConnectedPromise, fulfillCallConnectedPromise, rejectCallConnectedPromise) = Promise<Void>.pending()
         self.callConnectedPromise = callConnectedPromise
-        self.fulfillCallConnectedPromise = fulfillCallConnectedPromise
+        self.fulfillCallConnectedPromise = {
```
Comment (michaelkirk-signal) [inline Signal/src/call/CallService.swift]: #tuplegate =(

File: Signal/test/call/PeerConnectionClientTest.swift [hunk 3]
```diff
@@ -12,35 +12,37 @@ import WebRTC
...
     var dataChannelMessages = [OWSWebRTCProtosData]()
 
-    internal func peerConnectionClientIceConnected(_ peerconnectionClient: PeerConnectionClient) {
+    func peerConnectionClientIceConnected(_ peerconnectionClient: PeerConnectionClient) {
         connectionState = .connected
     }
 
-    internal func peerConnectionClientIceFailed(_ peerconnectionClient: PeerConnectionClient) {
+    func peerConnectionClientIceDisconnected(_ peerconnectionClient: PeerConnectionClient) {
+        connectionState = .disconnected
```
Comment (michaelkirk-signal) [inline Signal/test/call/PeerConnectionClientTest.swift]: Thanks for fixing up these tests.

File: Signal.xcodeproj/project.pbxproj (hunk 1 above)
Comment (charlesmchen-signal) [reply Signal.xcodeproj/project.pbxproj]: > SWIFT_SWIFT3_OBJC_INFERENCE

has been disabled everywhere.

> SWIFT_VERSION

So, if you select 4.1 in the XCode project/target build settings, the actual value that is persisted is 4.0.  So I think we're okay?

File: Signal/test/call/PeerConnectionClientTest.swift (hunk 3 above)
Comment (charlesmchen-signal) [reply Signal/test/call/PeerConnectionClientTest.swift]: I just pushed one more commit to get these tests not just building but passing.

https://github.com/signalapp/Signal-iOS/pull/3482/commits/f7abcc9064eecc911f3fe9d3f10606dd8dcadc35

**Pull request merged**